src/com/carolinarollergirls/scoreboard/defaults/DefaultScoreBoardModel.java

```diff
@@ -579,59 +522,121 @@ protected void createTeamModel(String id) {
+					ic.setMaximumTime(settings.getLong("Clock." + Clock.ID_INTERMISSION + ".Time"));
+					ic.resetTime();
+					ic.start();
+				}
+				requestBatchEnd();
 			}
-		};
+		}
+	};
 	protected ScoreBoardListener jamStartListener = new ScoreBoardListener() {
-			public void scoreBoardChange(ScoreBoardEvent event) {
-				ClockModel lc = getClockModel(Clock.ID_LINEUP);
-				if (lc.isRunning())
-					lc.stop();
-			}
-		};
-		
-		protected ScoreBoardListener jamEndListener = new ScoreBoardListener() {
-			public void scoreBoardChange(ScoreBoardEvent event) {
-				Clock j = getClock(Clock.ID_JAM);
-				if (event.getProvider() == j && !j.isRunning() && j.isTimeAtEnd()) {
-					_stopJam();
-				}
+		public void scoreBoardChange(ScoreBoardEvent event) {
+			ClockModel lc = getClockModel(Clock.ID_LINEUP);
+			if (lc.isRunning())
+				lc.stop();
+		}
+	};
+	protected ScoreBoardListener jamEndListener = new ScoreBoardListener() {
+		public void scoreBoardChange(ScoreBoardEvent event) {
+			ClockModel jc = getClockModel(Clock.ID_JAM);
+			if (jc.isTimeAtEnd()) {
+				_stopJam();
 			}
-		};
-		
+		}
+	};
 	protected ScoreBoardListener timeoutStartListener = new ScoreBoardListener() {
-			public void scoreBoardChange(ScoreBoardEvent event) {
-				ClockModel ic = getClockModel(Clock.ID_INTERMISSION);
-				Clock tc = getClock(Clock.ID_TIMEOUT);
+		public void scoreBoardChange(ScoreBoardEvent event) {
+			ClockModel ic = getClockModel(Clock.ID_INTERMISSION);
+			ClockModel tc = getClockModel(Clock.ID_TIMEOUT);
 
-				if (!isInPeriod() && ic.isRunning() && tc.isRunning()) {
-					ic.stop();
-					setInPeriod(true);
-				}
+			if (!isInPeriod() && ic.isRunning() && tc.isRunning()) {
+				ic.stop();
+				setInPeriod(true);
 			}
-		};
+		}
+	};
 	protected ScoreBoardListener timeoutEndListener = new ScoreBoardListener() {
-			public void scoreBoardChange(ScoreBoardEvent event) {
+		public void scoreBoardChange(ScoreBoardEvent event) {
+			requestBatchStart();
+			setTimeoutOwner("");
+			setOfficialReview(false);
+			getClockModel(Clock.ID_TIMEOUT).changeNumber(1);
+			requestBatchEnd();
+		}
+	};
+	protected ScoreBoardListener intermissionEndListener = new ScoreBoardListener() {
+		public void scoreBoardChange(ScoreBoardEvent event) {
+			ClockModel ic = getClockModel(Clock.ID_INTERMISSION);
+			ClockModel pc = getClockModel(Clock.ID_PERIOD);
+			ClockModel jc = getClockModel(Clock.ID_JAM);
+			if (ic.getTimeElapsed() > ic.getTimeRemaining() && pc.getNumber() < pc.getMaximumNumber()) {
+				//if more than half of the intermission has elapsed and there is another period, 
+				// start the next period. Otherwise extend the previous period.
 				requestBatchStart();
-				setTimeoutOwner("");
-				setOfficialReview(false);
-				getClockModel(Clock.ID_TIMEOUT).changeNumber(1);
+				pc.changeNumber(1);
+				pc.resetTime();
+				if (settings.getBoolean("ScoreBoard." + Clock.ID_JAM + ".Number")) {
+					jc.setNumber(1);
+				} else {
+					jc.changeNumber(1);
```

Comment:
It allows to have jam numbers to continue counting through all periods instead of restarting at 1 each period.  (This functionality is moved over from policies.)

Reply:
That setting naming is a bit confusing then, as it overlaps with the WS event name for the jam number. Can you choose a different setting name?